projet_partner_bank_type/models/account_payment_line.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2025 - TODAY, Cristiano Mafra Junior <cristiano.mafra@escodoo.com.br>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class AccountPaymentLine(models.Model):
+    _inherit = "account.payment.line"
+
+    @api.onchange("partner_id")
+    def partner_id_change(self):
+        res = super().partner_id_change()
+
+        bank = False
+        journal = self.move_line_id and self.move_line_id.move_id.journal_id or False
+        if journal:
+            if journal.code == "SLR":
+                bank = self.partner_id.bank_ids.filtered(
+                    lambda b: b.bank_usage == "salary"
+                )[:1]
+            elif journal.code == "DES":
+                bank = self.partner_id.bank_ids.filtered(
+                    lambda b: b.bank_usage == "expense"
+                )[:1]
+
+        self.partner_bank_id = bank.id if bank else False
```

Comment:
Caso não tenha encontrado a conta, não alterar para `False`

```suggestion
        if bank:
            self.partner_bank_id = bank.id
```